### PR TITLE
Uninstall `ember-classic-decorator`

### PR DESCRIPTION
**Pull request by charlesfries**
`ember-classic-decorator` in its current state is not compatible with Ember 4 (the following PR will fix it but it's yet to be merged: https://github.com/emberjs/ember-classic-decorator/pull/83)

I think we are able to get by without it, but I'm going to do some research into Ember Data and Ember Simple Auth to make sure. Opening the PR anyways.

### CODE REVIEW

File: addon/adapters/cloud-firestore-modular.ts [hunk 1]
```diff
@@ -86,8 +84,9 @@ export default class CloudFirestoreModularAdapter extends Adapter {
     return fastboot && fastboot.isFastBoot;
   }
 
-  public init(...args: unknown[]): void {
-    this._super(...args);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public constructor(...args: any[]) {
```
Comment (charlesfries) [inline addon/adapters/cloud-firestore-modular.ts:88]: Tried using the `unknown[]` type here but I can't then call `super()` with that unknown type.

**Comment by mikkopaderes** [comment]
I can recall now why I used `ember-classic-decorator`, it's because if we use the `init` hook, the eslint will throw [this](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/classic-decorator-hooks.md) error

**Review by mikkopaderes — APPROVED** [review]

**Review by mikkopaderes — CHANGES_REQUESTED** [review]
Let me hold off on this change. I'm applying it in #217 as I want to fix all deprecations in 1 PR so that we could see if the CI build succeeds.

**Comment by mikkopaderes** [comment]
@charlesfries I've applied this in #217. Thanks for providing the fix!

**Pull request closed without merging**